nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -227,7 +248,7 @@ def training_step(self, batch, batch_idx):
         batch_for_pipeline = self.process_global_batch(batch)
         encoder_seq_length = batch_for_pipeline[0].size(1)
         decoder_seq_length = batch_for_pipeline[1].size(1)
-        tensor_shape = [encoder_seq_length, get_micro_batch_size(), self.cfg.hidden_size]
+        tensor_shape = [encoder_seq_length, get_micro_batch_size(), self.cfg.encoder.hidden_size]
```

Comment:
Do we allow encoder and decoder to have different size? If not, do we check for it during config validation stage?

Reply:
Yes we do check if hidden sizes are different and throw an error if they are. Checkout `_validate_config()` in `token_level_encoder_decoder.py`